eclair-core/src/main/scala/fr/acinq/eclair/io/Switchboard.scala

```diff
@@ -22,22 +22,34 @@ import akka.actor.{Actor, ActorLogging, ActorRef, OneForOneStrategy, Props, Stat
 import fr.acinq.bitcoin.Crypto.PublicKey
 import fr.acinq.eclair.NodeParams
 import fr.acinq.eclair.blockchain.EclairWallet
-import fr.acinq.eclair.channel.HasCommitments
+import fr.acinq.eclair.channel._
+import fr.acinq.eclair.payment.Relayed
 import fr.acinq.eclair.router.Rebroadcast
+import fr.acinq.eclair.transactions.OUT
+import fr.acinq.eclair.wire.{TemporaryNodeFailure, UpdateAddHtlc}
+import grizzled.slf4j.Logging
 
 /**
   * Ties network connections to peers.
   * Created by PM on 14/02/2017.
   */
 class Switchboard(nodeParams: NodeParams, authenticator: ActorRef, watcher: ActorRef, router: ActorRef, relayer: ActorRef, wallet: EclairWallet) extends Actor with ActorLogging {
 
+  import Switchboard._
+
   authenticator ! self
 
   // we load peers and channels from database
   private val initialPeers = {
-    val channels = nodeParams.channelsDb.listChannels().groupBy(_.commitments.remoteParams.nodeId)
+    val channels = nodeParams.channelsDb.listChannels()
     val peers = nodeParams.peersDb.listPeers()
+    
+    val brokenHtlcs = checkBrokenHtlcsLink(channels)
+    val brokenHtlcKiller = context.actorOf(Props[HtlcReaper], name = "htlc-reaper")
+    brokenHtlcKiller ! brokenHtlcs
```

Comment:
Since broken HTLCs are known when we create the reaper actor, why not pass them to its constructor ?

Reply:
Because they would never be garbage collected.

Reply:
It would also be a bit more efficient not to create the reaper actor if the list is empty

Reply:
I did it, but I don't really like that kind of branching that early in the code.

Also, the fact that the reaper's state is never cleaned up is not great.